test/check_sharding_data_test.lua

```diff
@@ -219,3 +268,30 @@ function g.test_invalid_key_reference_path()
     )
     t.assert_equals(ok, nil)
 end
+
+function g.test_invalid_sharding_func()
+    -- unknown name
+    g.space.sharding_func = 'bad_sharding_func'
+    local ok, err = ddl.check_schema(g.schema)
+    t.assert_equals(err,
+        [[spaces["space"].sharding_func: unknown sharding_func "bad_sharding_func"]]
+    )
+    t.assert_equals(ok, nil)
+
+    -- invalid type
+    g.space.sharding_func = 5
+    local ok, err = ddl.check_schema(g.schema)
+    t.assert_equals(err,
+        [[spaces["space"].sharding_func: bad value (string or table expected, got number)]]
+    )
+    t.assert_equals(ok, nil)
+
+    -- sharding_func without body
+    g.space.sharding_func = {}
+    local ok, err = ddl.check_schema(g.schema)
+    t.assert_equals(err,
+        [[spaces["space"].sharding_func: sharding_func exists in the space as table,]] ..
+        [[but there's no 'body' field in sharding_func section]]
+    )
+    t.assert_equals(ok, nil)
```

Comment:
I would split test into a number of tests.  `assert()` failed in the beginning of test will block other testcases.

Reply:
Done.